### PR TITLE
fix(client): use babel polyfill for async

### LINKED ISSUE #34572: Code with async / await will not run
**Describe the bug**
Even though we use Babel to transform code that the student enters, it doesn't support `async` and `await`. This is also true of code added to the `head` or `tail` sections of the curriculum.

**To Reproduce**
1. Locally use the latest development version of the "learn" platform with the Monaco editor (not the production version).
2. Go to any lesson where you can enter javascript. This lesson is fine. http://localhost:8000/learn/front-end-libraries/react/create-a-simple-jsx-element
3. Enter the code below into the editor.

```
console.log('Gets here 0');

function later() {
    return new Promise(function(resolve) {
        setTimeout(resolve, 100);
    });
}

console.log('Gets here 1');

async function thisIsOkUntilYouRunIt() {
  console.log('Before await: thisIsOkUntilYouRunIt. Will not get here.');
  await later();
  console.log('After await: thisIsOkUntilYouRunIt. Will not get here.');
}

console.log('Gets here 2');

thisIsOkUntilYouRunIt();

console.log('Will not get here.');
```

4. Look at the console to see the code does not finished executing, per the console messages.

**Expected behavior**
The code should finish running: `async` and `await` should be supported. In fact, it is the Babel transform that is causing the problem. The above would run in most modern browsers.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Browser Chrome and Firefox, latest stable versions.

**Additional context**
Here is the code as transformed by Babel, which helps explain the problem.

```
"use strict";
;/*fcc*/
"use strict";

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

var JSX = React.createElement("div", null);
console.log('Gets here 0');

function later() {
  return new Promise(function (resolve) {
    setTimeout(resolve, 100);
  });
}

console.log('Gets here 1');

function thisIsOkUntilYouRunIt() {
  return _thisIsOkUntilYouRunIt.apply(this, arguments);
}

function _thisIsOkUntilYouRunIt() {
  _thisIsOkUntilYouRunIt = _asyncToGenerator(
  /*#__PURE__*/
  regeneratorRuntime.mark(function _callee() {
    return regeneratorRuntime.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            console.log('Before await: thisIsOkUntilYouRunIt. Will not get here.');
            _context.next = 3;
            return later();

          case 3:
            console.log('After await: thisIsOkUntilYouRunIt. Will not get here.');

          case 4:
          case "end":
            return _context.stop();
        }
      }
    }, _callee, this);
  }));
  return _thisIsOkUntilYouRunIt.apply(this, arguments);
}

console.log('Gets here 2');
thisIsOkUntilYouRunIt();
console.log('Will not get here.');
;/*fcc*/
```
If you paste the code above directly into the console (making sure you select the correct iframe to run it in), then you will get the following error:

```
ReferenceError: regeneratorRuntime is not defined
```
What it seems like is happening, is that Babel is transforming the code to run in any browser, but the iframe doesn't have all the parts of Babel that it needs in order to run that transformed code.

This issue on Stackoverflow might help. https://stackoverflow.com/questions/28976748/regeneratorruntime-is-not-defined

Do we need to add Babel polyfill to that iframe? I'm not sure how to do that even. I've tried a few things and none of them have worked. I can try to submit a PR for this if someone can point me in the right direction about what to do, and where to do it.

### PR DESCRIPTION
**Pull request by ValeraS**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.

Closes #34572

### CODE REVIEW

**Review by raisedadead — COMMENTED** [review]
Nice work, we do want to get away from Cloudflare though. This is because our client app is getting deployed to Netlify. They have their own CDN. 

There are two approaches:

1. Bundle this into the client app. (Potentially increase the build size and time - minor though)
2. Serve these binaries off our [CDN repo](https://github.com/freeCodeCamp/cdn). This is deployed to Netlify at `https://cdn.freecodecamp.org`

**Comment by ValeraS** [comment]
1. We can't bundle this polyfill. @Bouncey can correct me if I'm wrong.
2. Could we do it in another PR with other links?

The client app uses the next non-freeCodeCamp links now:
- https://cdnjs.cloudflare.com/ajax/libs/sass.js/0.10.11/sass.sync.min.js
- https://cdnjs.cloudflare.com/ajax/libs/chai/4.2.0/chai.min.js
- https://cdnjs.cloudflare.com/ajax/libs/d3/5.7.0/d3.min.js
- https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-AMS_HTML
- https://cdn.jsdelivr.net/npm/jquery@3.3.1/dist/jquery.min.js
- https://cdnjs.cloudflare.com/ajax/libs/normalize/4.2.0/normalize.min.css
- https://cdnjs.cloudflare.com/ajax/libs/babel-polyfill/7.0.0/polyfill.min.js <-- added with this PR

additionally from challenge files:
- https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.2.0/css/font-awesome.css
- https://cdnjs.cloudflare.com/ajax/libs/animate.css/3.2.0/animate.css
- https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.css
- https://unpkg.com/react@16.4.0/umd/react.production.min.js
- https://unpkg.com/react-dom@16.4.0/umd/react-dom.production.min.js
- https://cdnjs.cloudflare.com/ajax/libs/redux/3.7.2/redux.min.js
- https://cdnjs.cloudflare.com/ajax/libs/react-redux/5.0.7/react-redux.min.js

File: client/src/templates/Challenges/rechallenge/builders.js [hunk 1]
```diff
@@ -101,13 +101,18 @@ A required file can not have both a src and a link: src = ${src}, link = ${link}
     .map(source => createBody({ source }))
   );
 
+  /* eslint-disable max-len */
+  const babelPolyfillCDN =
+    'https://cdnjs.cloudflare.com/ajax/libs/babel-polyfill/7.0.0/polyfill.min.js';
+  const babelPolyfill = `<script src="${babelPolyfillCDN}" type="text/javascript"></script>`;
```
Comment (tchaffee) [inline client/src/templates/Challenges/rechallenge/builders.js]: I came SO close to figuring this solution out myself but couldn't quite get it. Thanks @ValeraS . Good work.

**Comment by raisedadead** [comment]
> Could we do it in another PR with other links?

YES PLEASE!

Thanks a lot for taking the time to find this comprehensive list.

**Review by Bouncey — APPROVED** [review]

**Pull request merged**